frontend/beCompliant/src/pages/LockedCreateContextPage.tsx

```diff
@@ -0,0 +1,150 @@
+import { UserInfo } from '../hooks/useFetchUserinfo';
+import {
+  Box,
+  Button,
+  Flex,
+  FormControl,
+  Select,
+  Spinner,
+  Stack,
+  Text,
+} from '@kvib/react';
+import { Form, useNavigate, useSearchParams } from 'react-router-dom';
+import { useSubmitContext } from '../hooks/useSubmitContext';
+import { FormEvent, useCallback, useState } from 'react';
+import { Table } from '../api/types';
+
+type Props = {
+  userinfo: UserInfo;
+  tablesData: Table[];
+};
+
+export const LockedCreateContextPage = ({ userinfo, tablesData }: Props) => {
+  const [search, setSearch] = useSearchParams();
+  const navigate = useNavigate();
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const teamId = search.get('teamId');
+  const name = search.get('name');
+  const tableId = search.get('tableId');
+  const redirect = search.get('redirect');
+
+  const setTableId = useCallback(
+    (newTableId: string) => {
+      search.set('tableId', newTableId);
+      setSearch(search);
+    },
+    [search, setSearch]
+  );
+
+  const { mutate: submitContext } = useSubmitContext();
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (teamId && name && tableId) {
+      setIsSubmitting(true); // Start submission
+      submitContext(
+        { teamId, tableId, name },
+        {
+          onSuccess: (data) => {
+            setIsSubmitting(false);
+            if (redirect) {
+              const incomingRedirect = decodeURIComponent(redirect)
+                .replace('{contextId}', data.data.id)
+                .replace('{contextName}', data.data.name)
+                .replace(
+                  '{tableName}',
+                  tablesData?.find((table) => table.id === data.data.tableId)
+                    ?.name ?? tableId
+                );
+              const fullRedirect = new URL(incomingRedirect);
+              const newRedirect = new URL(
+                `${window.location.origin}/context/${data.data.id}`
+              );
+              fullRedirect.searchParams.set(
+                'redirect',
+                `${newRedirect.toString()}`
+              );
+              window.location.href = fullRedirect.toString();
+            } else {
+              navigate(`/context/${data.data.id}`);
+            }
+          },
+          onError: () => {
+            setIsSubmitting(false);
+          },
+        }
+      );
+    } else {
+      console.error('teamId, tableId, and contextName must be provided');
+    }
+  };
+
+  const isButtonDisabled = !teamId || !tableId || !name || isSubmitting;
+
+  const getTeamDisplayName = (teamId: string | null) => {
+    return userinfo.groups.find((group) => group.id === teamId)?.displayName;
+  };
```

Comment:
Dette trenger kanskje ikke være en funksjon, ettersom vi allerede har `teamId` i komponentet

```suggestion
  const teamDisplayName = userinfo.groups.find((group) => group.id === teamId)?.displayName;
```